### PR TITLE
[Personalize] Update `friendlyId` format

### PR DESCRIPTION
**Pull request by sc-addypathania**
Updated friendlyId format from `<id>_<language>_<version>`  to `embedded_<id>_<language>`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/sitecore-jss/src/personalize/graphql-personalize-service.ts [hunk 1]
```diff
@@ -97,7 +99,7 @@ export class GraphQLPersonalizeService {
       return data?.layout?.item
         ? {
             // CDP expects content id format `<id>_<language>_<version>` (lowercase)
-            contentId: `${data.layout.item.id}_${language}_${data.layout.item.version}`.toLowerCase(),
+            contentId: `${EMBEDDED}${data.layout.item.id}_${language}`.toLowerCase(),
```
Comment (illiakovalenko) [inline packages/sitecore-jss/src/personalize/graphql-personalize-service.ts:102]: Probably you can just write `embedded${data.layout.item.id}...` - anyway this prefix is not customizable, may be not needed to have a separate variable

Comment (illiakovalenko) [inline packages/sitecore-jss/src/personalize/graphql-personalize-service.ts:102]: And `_` is missing after `embedded`
```suggestion
            contentId: `embedded_${data.layout.item.id}_${language}`.toLowerCase(),
```

Comment (illiakovalenko) [inline packages/sitecore-jss/src/personalize/graphql-personalize-service.ts:102]: You need to reflect same changes for `personalize-middleware` tests - https://github.com/Sitecore/jss/blob/dev/packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.test.ts

File: packages/sitecore-jss/src/personalize/graphql-personalize-service.test.ts [hunk 2]
```diff
@@ -48,7 +49,7 @@ describe('GraphQLPersonalizeService', () => {
     const personalizeData = await service.getPersonalizeInfo('/sitecore/content/home', 'en');
 
     expect(personalizeData).to.deep.equal({
-      contentId: `${id}_en_${version}`.toLowerCase(),
+      contentId: `${EMBEDDED}${id}_en`.toLowerCase(),
```
Comment (illiakovalenko) [inline packages/sitecore-jss/src/personalize/graphql-personalize-service.test.ts:52]:
```suggestion
      contentId: `embedded_${id}_en`.toLowerCase(),
```

**Review by illiakovalenko — CHANGES_REQUESTED** [review]

File: packages/sitecore-jss/src/personalize/graphql-personalize-service.ts (hunk 1 above)
Comment (sc-addypathania) [reply packages/sitecore-jss/src/personalize/graphql-personalize-service.ts:102]: Yeah make sense. Thanks!

File: packages/sitecore-jss/src/personalize/graphql-personalize-service.ts [hunk 3]
```diff
@@ -97,7 +97,7 @@ export class GraphQLPersonalizeService {
       return data?.layout?.item
         ? {
             // CDP expects content id format `<id>_<language>_<version>` (lowercase)
```
Comment (ambrauer) [inline packages/sitecore-jss/src/personalize/graphql-personalize-service.ts:99]: Update the comment to match new format

File: packages/sitecore-jss/src/personalize/graphql-personalize-service.ts [hunk 4]
```diff
@@ -97,7 +97,7 @@ export class GraphQLPersonalizeService {
...
-            contentId: `${data.layout.item.id}_${language}_${data.layout.item.version}`.toLowerCase(),
+            contentId: `embedded_${data.layout.item.id}_${language}`.toLowerCase(),
```
Comment (ambrauer) [inline packages/sitecore-jss/src/personalize/graphql-personalize-service.ts:100]: According to the spec - on language, we also need to replace dashes(-) with underscores(_). We should add a unit test for this case too.

**Review by ambrauer — CHANGES_REQUESTED** [review]

**Review by ambrauer — APPROVED** [review]
Looks good

**Pull request merged**